### PR TITLE
`make all` fails unless you first run `make tools`

### PR DESCRIPTION
**Pull request by drewmullen**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the Cloudformation Open Coverage Roadmap.

### Background

`make all` failed at documentation build because i had not run `make tools`. 

solved by updating `make all` to include `tools`. I also change `build` to be `install` and made `build` behave how `all` used to behave.

### CODE REVIEW

File: contributing/DEVELOPMENT.md [hunk 1]
```diff
@@ -56,7 +50,7 @@ $ make testacc
 
 [Development overrides for provider developers](https://www.terraform.io/docs/cli/config/config-file.html#development-overrides-for-provider-developers) can be leveraged in order to use the provider built from source.
 
-To do this, populate a Terraform CLI configuration file (`~/.terraformrc` for all platforms other than Windows; `terraform.rc` in the `%APPDATA%` directory when using Windows) with at least the following options:
+To do this, populate a Terraform CLI configuration file (`~/.terraformrc` for all platforms other than Windows; `terraform.rc` in the `%APPDATA%` directory when using Windows) with at least the below options and also set `TF_CLI_CONFIG_FILE=$HOME/.terraformrc`.
```
Comment (ewbankkit) [inline contributing/DEVELOPMENT.md:53]: Please add a link to the documentation: https://www.terraform.io/docs/cli/config/config-file.html
Also, setting `TF_CLI_CONFIG_FILE` shouldn't be necessary if the configuration file is correctly located for your platform (I just verified this on mac).

Comment (drewmullen) [reply contributing/DEVELOPMENT.md:53]: this seems to be a bug: https://github.com/hashicorp/terraform/issues/30169

I will update the docs to remove that env var

File: GNUmakefile [hunk 2]
```diff
@@ -4,13 +4,15 @@ PKG_NAME=internal/aws/...
...
 ACCTEST_PARALLELISM?=20
 
-default: build
+default: install
 
-.PHONY: all build default docs golangci-lint lint plural-data-sources resources schemas singular-data-sources test testacc tools
+.PHONY: all install default docs golangci-lint lint plural-data-sources resources schemas singular-data-sources test testacc tools
 
-all: schemas resources singular-data-sources plural-data-sources build docs
+all: tools schemas resources singular-data-sources plural-data-sources install docs
```
Comment (gdavison) [inline GNUmakefile:11]: This is equivalent to `tools build`. Would it make sense to change it, so if `build` gets a new step then `all` would pick it up?

**Review by gdavison — APPROVED** [review]
Looks good, @drewmullen. Just one question

**Pull request closed without merging**